capstone-app/src/app/modules/photos/components/photos-section/photos-section.component.ts

```diff
@@ -1,17 +1,111 @@
-import { Component } from '@angular/core';
+import {
+  Component,
+  Input,
+  OnChanges,
+  ChangeDetectorRef,
+  AfterViewChecked,
+} from '@angular/core';
 import { FormControl, Validators } from '@angular/forms';
+import MockPlacesPa from 'src/app/modules/photos/assets/mock-places-phoenixville.json';
+import MockPlacesBos from 'src/app/modules/photos/assets/mock-places-boston.json';
+import { NavItem } from '../../nav-item';
 
 @Component({
   selector: 'app-photos-section',
   templateUrl: './photos-section.component.html',
-  styleUrls: ['./photos-section.component.scss']
+  styleUrls: ['./photos-section.component.scss'],
 })
-export class PhotosSectionComponent {
+export class PhotosSectionComponent implements OnChanges, AfterViewChecked {
   limitControl = new FormControl(10, [Validators.min(1), Validators.max(10)]);
   maxPhotos = 10;
+  mockPlacesPa: any = (MockPlacesPa as any).places;
+  mockPlacesBos: any = (MockPlacesBos as any).places;
+  mockPlaces: any;
+  /**
+   * These variables are meant to detect the categories markers belong to.
+   * uniqueTypes: Some markers belong to more than one category.
+   * allTypes: The types before extracting the unique ones
+   */
+  uniqueTypes: string[];
+  allTypes: string[];
+  navItems: NavItem[] = [];
+  markerFilter = '';
+  @Input() city: string;
 
-  updateLimit(newLimit: number) {
-    this.maxPhotos = newLimit;
-    this.limitControl = new FormControl(newLimit);
+  constructor(private cdr: ChangeDetectorRef) {}
+
+  // Until the photos section is connected to the maps, only mocks are supported
+  ngOnChanges() {
+    if (this.city.includes('Boston')) {
+      this.mockPlaces = this.mockPlacesBos;
+    } else {
+      this.mockPlaces = this.mockPlacesPa;
+    }
+    this.extractUniqueTypes();
+    this.populateNavItems();
+  }
+
+  ngAfterViewChecked() {
+    this.cdr.detectChanges();
```

Comment:
Very frightened of this line. It seems like this could have massive performance consequences. Why is it necessary? Manual change detection is a sword without a hilt, there is no safe way to grasp it.

Reply:
I have this line because without it, I get a "Expressionhaschangedaftervaluehasbeenread" error

Reply:
Should I remove it? The code functions fine but the console throws that error

Reply:
What would be ideal is if you could track down what's causing the error, and find a way to eliminate without manually causing change detection. If that can't be done, then I think I'd rather have an error show up in the console than have manual change detection run in a life cycle method, due to the potential for performance issues (I think it might even be possible for this to lead to an infinite recursion). If you have to leave in the console error, I'd call it out with a comment or a TODO.

Reply:
All set actually. The error was occurring earlier on but since then I've changed some things about what was causing it and now I can safely remove this line without error.